File: js/factorial.js
exports.factorial = function(num) {
  if (typeof(num) !== 'number') {
    return null
  }
  if (num % 1 !== 0) {
    return null
  }
  if (num < 0) {
    return null
  }

  let product = 1
  for (let i = 2; i <= num; i++) {
    product *= i
  }

  return product
};

exports.bigFactorial = function(num) {
  if (typeof(num) !== 'bigint') {
    return null
  }
  if (num % 1n !== 0n) {
    return null
  }
  if (num < 0n) {
    return null
  }
  
  let product = BigInt(1)
  for (let i = num; i > BigInt(1); i--) {
    product *= i
  }

  console.log(product)
  return product
};

File: python/factorial.py
def factorial(num):
	# if (not isinstance(num, int)):
	# 	return('Number must be an integer greater than or equal to 0.')
	# elif(num < 0):
	# 	return('Number must be an integer greater than or equal to 0.')
	# answer = 1
	# while(num>0):
	# 	answer *= num
	# 	num -= 1
	# return(answer)
	if (not(type(num) == int) or num < 0):
		return(f'Input must be an integer greater than or equal to 0.')
	if num == 1:
		return num
	else:
		return num * factorial(num-1)


